src/script/interpreter.cpp

```diff
@@ -1303,12 +1303,12 @@ class CTransactionSignatureSerializer
         it = itBegin;
         while (scriptCode.GetOp(it, opcode)) {
             if (opcode == OP_CODESEPARATOR) {
-                s.write((char*)&itBegin[0], it-itBegin-1);
+                s.write(reinterpret_cast<const char*>(&itBegin[0]), it-itBegin-1);
```

Comment:
This is already fixed in https://github.com/bitcoin/bitcoin/pull/23438/files#diff-a0337ffd7259e8c7c9a7786d6dbd420c80abfa1afdb34ebae3261109d9ae3c19